src/index.js

```diff
@@ -11,26 +11,18 @@ import { assign } from './util';
  *    store.setState({ c: 'd' });   // logs { a: 'b', c: 'd' }
  */
 export default function createStore(state) {
-	let listeners = [];
+	const listeners = [];
 	state = state || {};
 
 	function unsubscribe(listener) {
-		let out = [];
-		for (let i=0; i<listeners.length; i++) {
-			if (listeners[i]===listener) {
-				listener = null;
```

Comment:
Looking at this, I'm not sure what value is added by setting `listener` to null here. Wouldn't `listener` just fall out of scope and be gc'ed anyway?

Reply:
Setting `listener` to `null` prevents the removal of any additional listeners in that loop.